### PR TITLE
Implement zone.on(event, callback).

### PR DESCRIPTION
**Pull request by zhanghuanrong**

### CODE REVIEW

File: inc/napa/capi.h [hunk 1]
```diff
@@ -92,6 +94,11 @@ EXTERN_C NAPA_API void napa_zone_execute(
 /// </remarks>
 EXTERN_C NAPA_API void napa_zone_recycle(napa_zone_handle handle);
 
+
+// <summary>
+EXTERN_C NAPA_API void napa_zone_on(napa_zone_handle handle, void* zoneEmitContxt);
```
Comment (helloshuangzi) [inline inc/napa/capi.h]: zoneEmitContext

File: lib/zone/zone-impl.ts [hunk 2]
```diff
@@ -184,4 +185,12 @@ export class ZoneImpl implements zone.Zone {
             transportContext: transportContext
         };
     }
+
+    public on(event: string, func: any) {
```
Comment (helloshuangzi) [inline lib/zone/zone-impl.ts]: >func: any [](start = 29, length = 9)

func: (...args: any[]) => any 

Use this to save the logic of checking type of func below?

Comment (zhanghuanrong) [reply lib/zone/zone-impl.ts]: good suggestion, fixed.

**Comment by helloshuangzi** [comment]
tracked by https://github.com/helloshuangzi/napajs/pull/20

**Pull request closed without merging**